### PR TITLE
vet: various cleanups

### PR DESCRIPTION
**Pull request by dfawley**
Various `vet.sh` improvements/cleanups:
- Use staticcheck for pkg comments instead of special code.
- Remove golint use (it is deprecated & very slow).
- Re-add deprecation warnings and update exclusions.
- Add all checkers to staticcheck (e.g. mismatched receivers, underscores).
- Add a manual check for misspelled test functions that prevent running.

And some fixes:
- Mismatched receiver names
- Test function misspelled that prevented it from running
- Duplicate imports

RELEASE NOTES: n/a

### CODE REVIEW

**Review by arvindbr8 — COMMENTED** [review]
only 2 minor comments; 

Also, does vet complain about `go.work` for you?

File: test/end2end_test.go [hunk 1]
```diff
@@ -775,8 +775,8 @@ func (te *test) configDial(opts ...grpc.DialOption) ([]grpc.DialOption, string)
 	}
 	if te.clientNopCompression {
 		opts = append(opts,
-			grpc.WithCompressor(NewNopCompressor()),
-			grpc.WithDecompressor(NewNopDecompressor()),
+			grpc.WithCompressor(newNopCompressor()),
```
[the remark this answers is not in the document]
Comment (dfawley) [reply test/end2end_test.go:778]: I'd rather not make changes that could cascade into more and more changes.  `WithCompressor` and `UseCompressor` are different in various ways.

File: test/end2end_test.go [hunk 2]
```diff
@@ -775,8 +775,8 @@ func (te *test) configDial(opts ...grpc.DialOption) ([]grpc.DialOption, string)
...
+			grpc.WithDecompressor(newNopDecompressor()),
```
[the remark this answers is not in the document]
Comment (dfawley) [reply test/end2end_test.go:779]: Same

File: vet.sh
[the remark this answers is not in the document]
Comment (dfawley) [reply vet.sh:1]: > I sometimes find it hard to stack trace when vet fails. Should we add somthing like `PS4='${LINENO}: '`? WDYT?

I'm not sure.  If you can improve things and show before/after output, then I can review your PR. :)

[the remark this answers is not in the document]
Comment (dfawley) [reply vet.sh:1]: > Also did you have trouble running with `go.work` file(s) when running `vet.sh`. Do you know of an easy way to get around it? Or should we add something in here to ignore `go.work` files?

Hmm, maybe a future work item would be to ignore files that are not in the git repo at all?

**Review by arvindbr8 — APPROVED** [review]

**Pull request merged**